server/tserver/src/main/java/org/apache/accumulo/tserver/FileManager.java

```diff
@@ -287,7 +290,15 @@ private void closeReaders(Collection<FileSKVIterator> filesToClose) {
     Map<FileSKVIterator,String> readersReserved = new HashMap<>();
 
     if (!tablet.isMeta()) {
+      long start = System.currentTimeMillis();
       filePermits.acquireUninterruptibly(files.size());
```

Comment:
Could use the tryAcquire method on the semaphore.  Something like

```java
      boolean acquired = filePermits.tryAcquire(files.size(), slowFilePermitMillis, TimeUnit.MILLISECONDS);
      if(!acquired) {
        //log message
        filePermits.acquireUninterruptibly(files.size());
      }
```

It would make the code slightly more complicated

Reply:
I initially implemented it that way, but the code bloat was a bit larger than that. tryAcquire returns false after the timeout so a while loop has to be used, time has to be aggregated over all the loops, and interrupts have to be handled. Due to the time aggregating I found myself using a start/stop time in addition to the timeout time for total wait time anyway.

Reply:
> I initially implemented it that way, but the code bloat was a bit larger than that.

I agree its not worth the bloat because I can't think of any strong need for immediacy.